### PR TITLE
textarea upgrade

### PR DESCRIPTION
**Pull request by matikrk**
* custom component Type

### CODE REVIEW

File: src/components/form-elements/Textarea.jsx [hunk 1]
```diff
@@ -38,11 +39,11 @@ const Textarea = props => {
     'sg-textarea--auto-height': autoHeight
   }, className);
 
-  return <textarea className={textareaClass} value={value} {...additionalProps} />;
+  return <Type className={textareaClass} value={value} {...additionalProps} />;
 };
 
 Textarea.propTypes = {
-  type: PropTypes.string,
+  Type: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
```
Comment (vergilius) [inline src/components/form-elements/Textarea.jsx]: why capitalized and why func allowed?

File: src/components/form-elements/Textarea.jsx [hunk 2]
```diff
@@ -19,6 +19,7 @@ const Textarea = props => {
     autoHeight,
     value,
     className,
+    Type = 'textarea',
```
Comment (vergilius) [inline src/components/form-elements/Textarea.jsx]: shouldn't this be in some kind of object, like enum or something?

File: src/components/form-elements/Textarea.jsx (hunk 1 above)
Comment (matikrk) [reply src/components/form-elements/Textarea.jsx]: string = normal component
func = react class

capitalized take variable  if small create html tag

File: src/components/form-elements/Textarea.jsx (hunk 2 above)
Comment (matikrk) [reply src/components/form-elements/Textarea.jsx]: textarea it's default and we basically want only textarea or React class looking like textarea. Ex. react-textarea-autosize from npm :)

**Pull request merged**